tests/core/mempool/test_mempool_manager.py

```diff
@@ -0,0 +1,100 @@
+from __future__ import annotations
+
+import asyncio
+from typing import Any, Iterator, List, Optional
+
+import pytest
+import pytest_asyncio
+from blspy import G2Element
+
+from chia.consensus.block_record import BlockRecord
+from chia.consensus.default_constants import DEFAULT_CONSTANTS
+from chia.full_node.mempool_manager import MempoolManager
+from chia.types.blockchain_format.classgroup import ClassgroupElement
+from chia.types.blockchain_format.coin import Coin
+from chia.types.blockchain_format.program import Program
+from chia.types.blockchain_format.sized_bytes import bytes32, bytes100
+from chia.types.coin_record import CoinRecord
+from chia.types.coin_spend import CoinSpend
+from chia.types.condition_opcodes import ConditionOpcode
+from chia.types.spend_bundle import SpendBundle
+from chia.util.errors import ValidationError
+from chia.util.ints import uint8, uint32, uint64, uint128
+
+IDENTITY_PUZZLE = Program.to(1)
+IDENTITY_PUZZLE_HASH = IDENTITY_PUZZLE.get_tree_hash()
+
+TEST_COIN = Coin(IDENTITY_PUZZLE_HASH, IDENTITY_PUZZLE_HASH, uint64(1000000000))
+TEST_HEIGHT = uint32(1)
+TEST_TIMESTAMP = uint64(1616108400)
+TEST_BLOCK_RECORD = BlockRecord(
+    IDENTITY_PUZZLE_HASH,
+    IDENTITY_PUZZLE_HASH,
+    TEST_HEIGHT,
+    uint128(0),
+    uint128(0),
+    uint8(0),
+    ClassgroupElement(bytes100(b"0" * 100)),
+    None,
+    IDENTITY_PUZZLE_HASH,
+    IDENTITY_PUZZLE_HASH,
+    uint64(0),
+    IDENTITY_PUZZLE_HASH,
+    IDENTITY_PUZZLE_HASH,
+    uint64(0),
+    uint8(0),
+    False,
+    uint32(TEST_HEIGHT - 1),
+    TEST_TIMESTAMP,
+    None,
+    uint64(0),
+    None,
+    None,
+    None,
+    None,
+    None,
+)
+
+
+async def get_coin_record(_: bytes32) -> Optional[CoinRecord]:
+    return None
+
+
+@pytest.fixture(scope="module")
+def event_loop() -> Iterator[asyncio.AbstractEventLoop]:
+    loop = asyncio.get_event_loop()
+    yield loop
+
+
+@pytest_asyncio.fixture(scope="module")
+async def mempool_manager() -> MempoolManager:
+    mempool_manager = MempoolManager(get_coin_record, DEFAULT_CONSTANTS)
+    await mempool_manager.new_peak(TEST_BLOCK_RECORD, None)
+    return mempool_manager
+
+
+def spend_bundle_from_conditions(conditions: List[List[Any]]) -> SpendBundle:
+    solution = Program.to(conditions)
+    coin_spend = CoinSpend(TEST_COIN, IDENTITY_PUZZLE, solution)
+    return SpendBundle([coin_spend], G2Element())
+
+
+@pytest.mark.asyncio
+async def test_addition_amount(mempool_manager: MempoolManager) -> None:
+    # Negative amount
+    conditions = [[ConditionOpcode.CREATE_COIN, IDENTITY_PUZZLE_HASH, -1]]
+    sb = spend_bundle_from_conditions(conditions)
+    sb_name = sb.name()
+    with pytest.raises(ValidationError, match="Err.INVALID_CONDITION"):
```

Comment:
it's not obvious why this would trigger an `INVALID_CONDITION`. Shouldn't it be `COIN_AMOUNT_NEGATIVE`?

Reply:
I'm exploring that through https://github.com/Chia-Network/chia_rs/pull/97